Reduces scope of stats notebook to single comparison and moves multi-factorial comparison to further notebook (#24)
CambridgeCentreForProteomics/course_expression_proteomics: materials/08_statistical_analysis_all_stages.Rmd

:::

```{r, echo = FALSE, fig.align = "center", out.width = "90%"}
knitr::include_graphics("figs/flow_chart/flow_chart.010.png", error = FALSE)
```

::: callout-note
This workflow is an adjunct to the [Statistical analysis](./06_statistical_analysis.html) section which demonstrates how to perform single comparison statistical with `limma` (@Smyth2004). Please first read through that material, since it includes background explanations and clarifications which are not repeated here.
:::

```{r, eval=TRUE, include=FALSE}
library("QFeatures")
library("NormalyzerDE")
library("limma")
library("factoextra")
library("org.Hs.eg.db")
library("clusterProfiler")
library("enrichplot")
library("patchwork")
library("tidyverse")
library("pheatmap")
library("here")

load(here("materials/output/lesson04.rda"), verbose = TRUE)
```

## Differential expression analysis

Having cleaned our data, aggregated from PSM to protein level, completed
a log2 transformation and normalised the data, we are now ready to carry
out statistical analysis. 

The aim of this section is to answer the
question: *"Which proteins show a significant change in abundance
between the multiple cell cycle stages?"*.

> Our null hypothesis is: **(H0)** The change in abundance for a protein
> between cell cycle stages is zero.
> 
> Our alternative hypothesis is: **(H1)** The change in abundance for a
> protein between cell cycle stages is not zero.

Reviewer: I feel like this could be slightly more intuitive:

```suggestion
Our null hypothesis is: **(H0)** The mean protein abundance is the same across cell cycle stages.

Our alternative hypothesis is: **(H1)** The mean protein abundance differs across cell cycle stages.
```

It might be worth adding a note somewhere explicitly saying that the F test does not tell us which groups (cell cycle stages) are different from each other. Only that there is some difference due to cell cycle. However, we will later see how to extract pairwise comparisons between each cell cycle.

Author: Agreed on both points

We will use `limma` to perform the statistical tests using an **empirical Bayes-moderated linear model**. Please see the [Statistical analysis](./06_statistical_analysis.html) section for more details about `limma` and its application in quantitative proteomics. 

Depending on whether the linear model is used to perform single comparisons or multifactorial comparisons, the test statistic will either be a

> t-value or an F-value, respectively. Here, we are performing a multifactorial comparison, so we will obtain an F-value. 

Reviewer: Well, I guess you will do both. Because when you extract the pairwise comparisons you get a t-value.
Maybe we could say:

```suggestion
t-value or an F-value, respectively. Here, we will perform both, starting with assessing the overall effect of cell cycle.
```

Author: Good point indeed!

### What is an F-value?

An F-value is a parametric statistical value used to compare the mean
values of **three or more** groups. The F-value is the ratio of the between-group
variation (explained variance) and the within-group variance (unexplained variance).
The higher the F-value, the more significant the difference between the groups is.

### How is the p-value obtained?

A p-value may be obtained from an F-value by comparing the value against an F-distribution with the appropriate degrees of freedom.
    
-   **degrees of freedom** = the number of observations minus the number
    of independent variables in the model
-   **p-value** = the probability of achieving the F-value under the
    null hypothesis i.e., by chance
    
Here, we will perform a comparison between three groups (M phase, G1 phase and
desynchronised) for each protein and obtain an F-value and p-value for each protein.

First, we extract the protein assay data. We will remove the pre-treatment sample,
since this condition is not replicated and so not amenable to statistical testing.

> ```{r}

Author: I don't see the change here? 🧐

## Identify the columns for the Pre-treatment sample in the protein assay
pre_treatment_col_ix <- colData(cc_qf[["log_norm_proteins"]])$condition == 'Pre-treatment'

# Extract the protein assay from our QFeatures object, excluding the pre-treatment sample

> all_proteins <- cc_qf[["log_norm_proteins"]][, !pre_treatment_col_ix]

Reviewer: Same comment as in the other notebook about possibly simplifying the code here so it doesn't have as many chained `[[`, `[`, `$`

Author: Agreed. Be thankful I didn't do it in one line 😆

```{r}
## Format results
limma_results_all_contrasts <- topTable(
  fit = final_model,   
  coef = NULL, 
  adjust.method = "BH",    # Method for multiple hypothesis testing
  number = Inf) %>%        # Print results for all proteins
  rownames_to_column("Protein") 

## Verify
head(limma_results_all_contrasts)
```

### QC plots for statistical test assumptions

First we examine whether there is the expected relationship between abundance and variance, and that the trend line captures this.
```{r}
plotSA(fit = final_model,
       cex = 0.5,
       xlab = "Average log2 abundance")
```

Next, we check that the p-value distribution is as expected.
```{r}

limma_results_all_contrasts %>%
  as_tibble() %>%
  ggplot(aes(x = P.Value)) + 
  geom_histogram()
```

#### Interpreting the overall linear model output

Having checked that the model we fitted was appropriate for the data, we can now take a look at the results of our test

As we saw above, `topTable` will give us the overall output of our linear model. We previously used this function to generate our `limma_results_all_contrasts` without specifying any value for the `coef` argument.

```{r}
head(limma_results_all_contrasts)
```

> Interpreting the output of `topTable` for a multi-contrast model:

Reviewer: ```suggestion
Interpreting the output of `topTable` when testing multiple coefficients simultaneously:
```
